### PR TITLE
Push Notification Manager: registration crash

### LINKED ISSUE #7873: Running from device(for my first time) causes error in PushNotificationsManager.swift
Hi I’m trying to run the app from Xcode on my device for the first time.  I recently got it to build and run after having some entitlement issues.  I set the entitlements to automatic and changed all the entitlements to point to my correct bundle group.  I also had to edit the keychain access groups to a different value in each entitlement.  The problem is now I get this error.  The app looks like it logs me in, but I’ve never put in any credentials to get to my site, and it doesn’t look like my login because it displays “You don’t have any WordPress sites yet”.  I get the error below and Xcode stops at this method in PushNotificationsManager.swift.  It crashes at the last line here with a SIGABRT error.

Should I change any of the values in Constants.m?  Do I have to do more setup on developer.apple.com?  All I did on there was to setup some groups and app ids.  I also setup the certificates for Push notifications and under Push Nofications at developer.apple.com the values are listed as Enabled.

Xcode 9.0
Device running: 10.3.2
iPhone 5s
Running from latest commit in develop: 97cb23a

```
    func registerDeviceToken(_ tokenData: Data) {
        // We want to register Helpshift regardless so that way if a user isn't logged in
        // they can still get push notifications that we replied to their support ticket.
        HelpshiftCore.registerDeviceToken(tokenData)
```

CoreData: annotation:  Failed to load optimized model at path '/var/containers/Bundle/Application/4A8676CA-AC9B-4F70-998B-BAD2D7AFD8D3/WordPress.app/WordPress.momd/WordPress 66.omo'
2017-09-28 11:49:36:559 WordPress[862:176390] WPAnalytics session started
2017-09-28 11:49:36:588 WordPress[862:176390] Helpshift Disabled
2017-09-28 11:49:36:902 WordPress[862:176390] User-Agent set to: Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_2 like Mac OS X) AppleWebKit/603.2.4 (KHTML, like Gecko) Mobile/14F89 wp-iphone/8.4
2017-09-28 11:49:37.006924-0700 WordPress[862:176390] [Warning] <UIVisualEffectView 0x153d35fc0> is being asked to animate its opacity. This will cause the effect to appear broken until opacity returns to 1.
2017-09-28 11:49:37.048446-0700 WordPress[862:176390] [Warning] Splitview controller <UISplitViewControllerClassicImpl: 0x153e49010> is expected to have a view controller at index 0 before it's used!
2017-09-28 11:49:37.461479-0700 WordPress[862:176390] Warning: Unable to create restoration in progress marker file
2017-09-28 11:49:37.744743-0700 WordPress[862:176390] +[LOTShapeGroup shapeItemWithJSON:frameRate:compBounds:]: Warning: merge shape is not supported. name: Merge Paths 1
Error while removing Share Extension OAuth2 bearer token: Error Domain=SFHFKeychainUtilsErrorDomain Code=-25300 "(null)"
Error while removing Share Extension Username: Error Domain=SFHFKeychainUtilsErrorDomain Code=-25300 "(null)"
2017-09-28 11:49:37.828419-0700 WordPress[862:176390] *** Terminating app due to uncaught exception 'InitializeError', reason: 'HelpshiftCore.init not called'
*** First throw call stack:
(0x18f2a2fe0 0x18dd04538 0x100a9eb1c 0x100a9f4d4 0x1008c9238 0x1008c9f94 0x1000afdac 0x103565a50 0x103565a10 0x10356ab78 0x18f2510c8 0x18f24ece4 0x18f17eda4 0x190be8074 0x195439058 0x1000a8600 0x18e18d59c)
libc++abi.dylib: terminating with uncaught exception of type NSException

### PR DESCRIPTION
**Pull request by frosty**
Fixes #7873.

A couple of folks who are new to the project and don't have the Helpshift API key in their credential files have seen a crash when the app attempts to register a device token with Helpshift. Not sure if this is a regression, but this PR puts a quick check in place before making the call to `registerDeviceToken`.

To test:

* Ensure the device token is registered correctly when it should be, and the app doesn't crash when there's no Helpshift key present.

Needs review: @astralbodies it looks like you touched this most recently – would you mind?

### CODE REVIEW

File: WordPress/Classes/ViewRelated/NUX/LoginEmailViewController.swift [hunk 1]
```diff
@@ -376,7 +376,7 @@ extension LoginEmailViewController: GIDSignInDelegate {
 
         let alert = UIAlertController(title: "Login Success", message: "Google login succeeded. Actual wpcom account login yet to be implemented.", preferredStyle: .alert)
         alert.addAction(UIAlertAction(title: "Ok. Thanks.", style: .default, handler: { [weak self] (action) in
-            self?.dismiss(animated: true){}
+            self?.dismiss(animated: true) {}
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/LoginEmailViewController.swift]: 😂

👌

Comment (elibud) [reply WordPress/Classes/ViewRelated/NUX/LoginEmailViewController.swift]: 👏 🤣 👏

**Comment by astralbodies** [comment]
@DerCSpringer Do you want to also try this PR out?

**Comment by DerCSpringer** [comment]
Sure I quickly put in those changes and they work with the commit and I mentioned in my post, but I haven't tried with the current commit

File: WordPress/Classes/ViewRelated/NUX/LoginEmailViewController.swift (hunk 1 above)
Comment (frosty) [reply WordPress/Classes/ViewRelated/NUX/LoginEmailViewController.swift]: Thanks to the pre-commit hooks I literally couldn't commit my actual changes until I fixed that 🙃

**Pull request merged**